programs/survival/graves.sc

```diff
@@ -0,0 +1,419 @@
+
+                if(datafound==false,
+                    _delete_data(pos(block));
+                    run(str('setblock %d %d %d minecraft:air',pos(block):0, pos(block):1, pos(block):2)); //for some reason /set doesnt seem to work. I might be doing bonk stuff somewhere
+                );
+            );
+        ));
+
+
+        if(screen_property(deathCrate,'open'),
+            c_for(i=0,i<41,i+=1,
+                item=gdata:str(i);
+                inventory_set(deathCrate,i,item:1,item:0,item:2);
+            );       
+        );
+        if(!gdata:'opened',modify(player,'add_xp',global_gdata:str(pos(block)):'xp'));
+        gdata:'opened'=true;
+    );
+);
+
+_owner_clicked(player,block) -> (
+
+    gdata=global_gdata:str(pos(block));
+    itemCount=0;
+    c_for(i=0,i<inventory_size(player),i+=1,
+        item=global_gdata:str(pos(block)):str(i);
+        if(item!=null && inventory_get(player,i)==null,
+            inventory_set(player,i,item:1,item:0,item:2);
+            _delete_data_slot(gdata,str(i)),
+         // when something is in slot
+            item!=null && inventory_get(player,i)!=null,
+            itemCount+=1;
+        );
+    );
+
+    if(itemCount == 0,
+        modify(player,'xp_level',gdata:'xp_level');
+        modify(player,'xp_progress',gdata:'xp_progress');
+        set(pos(block),'air');
+        _delete_data(pos(block));
+        return();          
+    );
+
+    deathCrate= create_screen(player,'generic_9x5',str('%s\'s Death Crate',gdata:'name'),_(screen,player,action,data,outer(gdata),outer(block)) ->(
+
+            if(action=='pickup' || action== 'swap' || action=='quick_move' || action== 'throw' || action== 'pickup_all',
+                _delete_data_slot(gdata,data:'slot');
+                datafound=false;
+            c_for(i=0,i<41,i+=1,if(gdata:str(i)!=null, datafound=true));
+
+            if(datafound==false,
+            _delete_data(pos(block));
+            modify(player,'xp_level',gdata:'xp_level');
+            modify(player,'xp_progress',gdata:'xp_progress');
+            run(str('setblock %d %d %d minecraft:air',pos(block):0, pos(block):1, pos(block):2));
+            );
+            );
+
+        ));
+
+    if(screen_property(deathCrate,'open'),
+
+            c_for(i=0,i<41,i+=1,
+                item=gdata:str(i);
+                inventory_set(deathCrate,i,item:1,item:0,item:2);
+            );      
+        );
+);
+
+
+
+
+//Utility Functions
+
+_saveGData() -> (
+    delete_file('gdata','JSON');
+    write_file('gdata','JSON',global_gdata);
+);
+
+_saveSData() -> (
+    delete_file('settings','JSON');
+    write_file('settings','JSON',global_settings);
+);
+
+_delete_data(pos) -> (
+
+    delete(global_gdata,str(pos));
+    delete_file('gdata','JSON');
+    _saveGData();
+
+);
+
+_delete_data_slot(data,slotNum) -> (
+
+    delete(data,str(slotNum));
+    delete_file('gdata','JSON');
+    _saveGData();
+
+);
```

Comment:
```suggestion
__on_close()->(
    app_data = {
        'gdata'->global_gdata,
        'settings'->global_settings
    };
    store_app_data(encode_nbt(app_data, false))
);
```

TBH you can get rid of these functions (and all their mentions), cos you only rly need to save to disk in the `__on_close()` event which fires right as the server is closing (still test it tho ofc).

Note, change the false to true if it gives you trouble in saving, though to see that you will probably need the output log open. It's just cos im not sure if it will accept the way you store the player data. If it doesn't, I'll help you with it dw.